### PR TITLE
day code menu, alert, datetime

### PR DESCRIPTION
**Pull request by vinhcotam**
## Related Tickets
- [#Task xxxxx: Content](https://edu-redmine.sun-asterisk.vn/issues/xxxx)
## WHAT
- Code về menu, alert và datetime
## Evidence (Screenshot or Video)
image
image
image
image
image
image
image
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_alert_date_time)
+    }
+
+    fun onClickShowAlert(view: View?) {
+        val exAlertBuilder: AlertDialog.Builder = AlertDialog.Builder(this)
+        exAlertBuilder.setTitle(R.string.alert)
+        exAlertBuilder.setMessage(R.string.alert_click_message)
+        exAlertBuilder.setPositiveButton(R.string.ok_message) { _, _ ->
+            Toast.makeText(applicationContext, "Pressed OK", Toast.LENGTH_LONG).show()
```
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt]: extract tring

File: app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        exAlertBuilder.setNegativeButton(R.string.cancel_message) { _, _ ->
+            Toast.makeText(applicationContext, "Pressed Cancel", Toast.LENGTH_LONG).show()
```
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt]: .

File: app/src/main/res/menu/main_menu.xml [hunk 3]
```diff
@@ -3,5 +3,22 @@
...
+    <item
+        android:id="@+id/ex_5"
+        android:title="@string/slide8"/>
+    <item
+        android:id="@+id/ex_6_1"
+        android:title="@string/slide9_1"/>
+    <item
+        android:id="@+id/ex_6_2"
+        android:title="@string/slide9_2"/>
 </menu>
```
Comment (hungld-1302) [inline app/src/main/res/menu/main_menu.xml]: endline

File: app/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -30,5 +30,8 @@
             @style/TextAppearance.AppCompat.Display1
         </item>
     </style>
-
+    <style name="ShoppingCart">
+        <item name="android:src">@drawable/ic_shoping_cart</item>
+        <item name="android:textAppearance"/>
+    </style>
 </resources>
```
Comment (hungld-1302) [inline app/src/main/res/values/styles.xml]: endline

File: app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt (hunk 1 above)
Comment (hunggld) [inline app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt:22]: cho vào resource string nha

File: app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt (hunk 2 above)
Comment (hunggld) [inline app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt:25]: .

File: app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        exAlertBuilder.show();
+    }
+
+    fun showDatePicker(view: View?) {
+        val newFragment: DialogFragment = DatePickerFragment()
+        newFragment.show(supportFragmentManager, "datePicker")
```
Comment (hunggld) [inline app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt:32]: .

File: app/src/main/java/com/sun/android/ex_slide9/DroidCafe.kt [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            } else {
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
+            }
+            recreate()
+        } else if (item.itemId == R.id.ex_1) {
+            val intentEx1 = Intent(this, ActivitySendMessage::class.java)
+            startActivity(intentEx1)
+        } else if (item.itemId == R.id.ex_2) {
+            val intentEx2 = Intent(this, ImplicitIntents::class.java)
+            startActivity(intentEx2)
```
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/ex_slide9/DroidCafe.kt]: case when

File: app/src/main/res/menu/main_menu.xml [hunk 7]
```diff
@@ -3,5 +3,22 @@
...
+        android:title="@string/slide6"/>
+    <item
+        android:id="@+id/ex_5"
+        android:title="@string/slide8"/>
+    <item
+        android:id="@+id/ex_6_1"
+        android:title="@string/slide9_1"/>
+    <item
+        android:id="@+id/ex_6_2"
+        android:title="@string/slide9_2"/>
```
Comment (hungvv-1301) [inline app/src/main/res/menu/main_menu.xml:23]:
```suggestion
        android:title="@string/slide9_2"/>
        
```

File: app/src/main/res/values/styles.xml (hunk 4 above)
Comment (hungvv-1301) [inline app/src/main/res/values/styles.xml]:
```suggestion
</resources>

```

File: app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    fun processDatePickerResult(year: Int, month: Int, day: Int) {
+        val month = (month + 1).toString()
+        val day = day.toString()
+        val year = year.toString()
+        val dateMessage = "$day/$month/$year"
+        Toast.makeText(this, "Ngày: $dateMessage", Toast.LENGTH_SHORT).show();
```
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/ex_slide9/AlertDateTime.kt]: Extract string

**Pull request merged**